tests/unit/test_gitlab_http_methods.py

```diff
@@ -372,6 +372,63 @@ def response_callback(
     assert "http://example.com/api/v4/user/status" in error_message
 
 
+def test_http_request_on_409_resource_lock_retries(gl_retry):
+    url = "http://localhost/api/v4/user"
+    retried = False
+
+    def response_callback(
+        response: requests.models.Response,
+    ) -> requests.models.Response:
+        """We need a callback that adds a resource lock reason only on first call"""
+        nonlocal retried
+
+        if not retried:
+            response.reason = "Resource lock"
+
+        retried = True
+        return response
+
+    with responses.RequestsMock(response_callback=response_callback) as rsps:
+        rsps.add(
+            method=responses.GET,
+            url=url,
+            status=409,
+            match=helpers.MATCH_EMPTY_QUERY_PARAMS,
+        )
+        rsps.add(
```

Comment:
Not sure I understand what is going on here. Is this command overrides the one in line 392?

Reply:
This adds a second response to the `rsps` object which tracks our responses for this context. It ensures the first mocked response will be a 409 and the second a 200, mimicking a successful response after retry.

See https://github.com/getsentry/responses#main-interface, and as we are using a context manager, also https://github.com/getsentry/responses#responses-as-a-context-manager.

Reply:
Thanks. The `Responses` package is a new one for me.